### PR TITLE
Replace deprecated `imghdr` package with `filetype`

### PR DESCRIPTION
**Pull request by geigerzaehler**
The `imghdr` package from the Python standard library is [deprecated and will be removed in Python 3.13][1]. The deprecated PEP suggests packages filetype, puremagic, python-magic as replacements. I chose filetype because it has a [larger user-base][2] than puremagic and does not have a C dependency, like python-magic.

[1]: https://peps.python.org/pep-0594/#deprecated-modules
[2]: https://libraries.io/pypi/filetype

### CODE REVIEW

File: mediafile.py [hunk 1]
```diff
@@ -346,51 +346,15 @@ def _sc_encode(gain, peak):
...
-        return 'image/x-xbitmap'
-    else:
-        return 'image/x-{0}'.format(kind)
+    return filetype.guess_mime(data)
 
 
 def image_extension(data):
-    ext = _imghdr_what_wrapper(data)
+    ext = filetype.guess_extension(data)
     return PREFERRED_IMAGE_EXTENSIONS.get(ext, ext)
```
Comment (jackwilsdon) [inline mediafile.py:358]: Is `PREFERRED_IMAGE_EXTENSIONS` still needed? `filetype` already returns `jpg` for JPEG files:

```
$ python3 -c 'import filetype; print(filetype.guess("./only-magic-bytes.jpg").extension)'
jpg
```

Comment (geigerzaehler) [reply mediafile.py]: You’re right! Good catch

**Comment by jackwilsdon** [comment]
It seems like there are quite a few changes to the return values of `image_mime_type` and `image_extension` as a part of switching to `filetype` (images from https://github.com/python/cpython/tree/v3.12.7/Lib/test/imghdrdata):

|File|Before|After|Match|
|-|-|-|-|
|imghdrdata/python.exr|image/x-exr (exr)||:heavy_multiplication_x:|
|imghdrdata/python.png|image/png (png)|image/png (png)|:heavy_check_mark:|
|imghdrdata/python.xbm|image/x-xbitmap (xbm)||:heavy_multiplication_x:|
|imghdrdata/python.sgi|image/x-rgb (rgb)||:heavy_multiplication_x:|
|imghdrdata/python.ras|image/x-rast (rast)||:heavy_multiplication_x:|
|imghdrdata/python.jpg|image/jpeg (jpg)|image/jpeg (jpg)|:heavy_check_mark:|
|imghdrdata/python.pgm|image/x-portable-graymap (pgm)||:heavy_multiplication_x:|
|imghdrdata/python.webp|image/x-webp (webp)|image/webp (webp)|:heavy_multiplication_x:|
|imghdrdata/python.gif|image/gif (gif)|image/gif (gif)|:heavy_check_mark:|
|imghdrdata/python.bmp|image/bmp (bmp)|image/bmp (bmp)|:heavy_check_mark:|
|imghdrdata/python.pbm|image/x-portable-bitmap (pbm)||:heavy_multiplication_x:|
|imghdrdata/python.tiff|image/tiff (tiff)|image/tiff (tif)|:heavy_multiplication_x:|
|imghdrdata/python-raw.jpg|image/jpeg (jpg)|image/jpeg (jpg)|:heavy_check_mark:|
|imghdrdata/python.ppm|image/x-portable-pixmap (ppm)||:heavy_multiplication_x:|

The majority of unrecognized types are ones that are less likely to be used, but it seems like the file extension for `tif` has changed which isn't ideal. Maybe this needs releasing as a breaking change?

**Comment by theol0403** [comment]
For those looking to run beets in python3.13, the following command is a good workaround until this gets merged:

`pip install git+https://github.com/geigerzaehler/mediafile/@replace-imghdr`

**Comment by gbcox** [comment]
This should probably be released soon as it's breaking in F41.  In the meantime, I'm going to see if I can apply as a patch to the F41  version.

**Comment by jackwilsdon** [comment]
I've added a change to preserve the `tiff` extension for TIFF files, resulting in this matchup:

|File|Before|After|Match|
|-|-|-|-|
|imghdrdata/python.exr|image/x-exr (exr)||:heavy_multiplication_x:|
|imghdrdata/python.png|image/png (png)|image/png (png)|:heavy_check_mark:|
|imghdrdata/python.xbm|image/x-xbitmap (xbm)||:heavy_multiplication_x:|
|imghdrdata/python.sgi|image/x-rgb (rgb)||:heavy_multiplication_x:|
|imghdrdata/python.ras|image/x-rast (rast)||:heavy_multiplication_x:|
|imghdrdata/python.jpg|image/jpeg (jpg)|image/jpeg (jpg)|:heavy_check_mark:|
|imghdrdata/python.pgm|image/x-portable-graymap (pgm)||:heavy_multiplication_x:|
|imghdrdata/python.webp|image/x-webp (webp)|image/webp (webp)|:heavy_multiplication_x:|
|imghdrdata/python.gif|image/gif (gif)|image/gif (gif)|:heavy_check_mark:|
|imghdrdata/python.bmp|image/bmp (bmp)|image/bmp (bmp)|:heavy_check_mark:|
|imghdrdata/python.pbm|image/x-portable-bitmap (pbm)||:heavy_multiplication_x:|
|imghdrdata/python.tiff|image/tiff (tiff)|image/tiff (tiff)|:heavy_check_mark:|
|imghdrdata/python-raw.jpg|image/jpeg (jpg)|image/jpeg (jpg)|:heavy_check_mark:|
|imghdrdata/python.ppm|image/x-portable-pixmap (ppm)||:heavy_multiplication_x:|

I think this is good enough - the only mismatch is `image/x-webp` which isn't the correct MIME type for WebP anyway, and the formats we've lost support for aren't normally seen in media files to begin with.

**Comment by jackwilsdon** [comment]
Thanks @geigerzaehler!

**Pull request merged**